packages/rs-sdk/src/core/transaction.rs

```diff
@@ -135,58 +133,140 @@ impl Sdk {
-                        //           );
-                        //
-                        //     return Ok(asset_lock_proof);
-                        // }
-                    }
-                    _ => {
-                        tracing::debug!("received something else");
+                        let merkle_block =
+                            MerkleBlock::consensus_decode(&mut raw_merkle_block.as_slice())
+                                .map_err(|e| {
+                                    tracing::error!("can't decode merkle block: {}", e);
+
+                                    Error::CoreError(e.into())
+                                })?;
+
+                        let mut matches: Vec<Txid> = vec![];
+                        let mut index: Vec<u32> = vec![];
+
+                        merkle_block.extract_matches(&mut matches, &mut index)?;
+
+                        // Continue receiving messages until we find the transaction
+                        if !matches.contains(&transaction_id) {
+                            tracing::debug!(
+                                "merkle block doesn't contain the transaction, waiting for the next message"
+                            );
+
+                            continue;
+                        }
+
+                        tracing::debug!(
+                            "merkle block contains the transaction, obtaining core chain locked height"
+                        );
+
+                        // Wait until the block is chainlocked
+                        let mut core_chain_locked_height;
+                        loop {
+                            let GetTransactionResponse {
+                                height,
+                                is_chain_locked,
+                                ..
+                            } = self
+                                .execute(
+                                    GetTransactionRequest {
+                                        id: transaction_id.to_string(),
+                                    },
+                                    RequestSettings::default(),
+                                )
+                                .await?;
+
+                            core_chain_locked_height = height;
+
+                            if is_chain_locked {
+                                break;
+                            }
+
+                            tracing::trace!("the transaction is on height {} but not chainlocked. try again in 1 sec", height);
+
+                            sleep(Duration::from_secs(1)).await;
+                        }
+
+                        tracing::debug!(
+                            "the transaction is chainlocked on height {}, waiting platform for reaching the same core height",
+                            core_chain_locked_height
+                        );
+
+                        // Wait until platform chain is on the block's chain locked height
```

Comment:
You are only getting that the validator set is on the block's chain locked height. This will not guarantee that the future node you query is.

Reply:
Metadata contains committed core chain locked height. If this core chain locked height is committed then next block will be on the same height or higher.

Reply:
Yes, but check_tx checks against the current state, which might not have the core_chain_locked_height.

Reply:
This might happen in many other situations even now when you create preorder and then domain for DPNS or update document and provide new revision.. I will create a task for this issue